module/VuFind/src/VuFind/Auth/Shibboleth/Shibboleth.php

```diff
@@ -44,28 +47,125 @@
  * @author   Bernd Oberknapp <bo@ub.uni-freiburg.de>
  * @author   Demian Katz <demian.katz@villanova.edu>
  * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @author   Vaclav Rosecky <vaclav.rosecky@mzk.cz>
  * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
  * @link     https://vufind.org Main Page
  */
 class Shibboleth extends AbstractBase
 {
+    /**
+     * Header name for entityID of the IdP that authenticated the user.
+     */
     const DEFAULT_IDPSERVERPARAM = 'Shib-Identity-Provider';
 
+    /**
+     * Header name of the internal session key assigned to the
+     * session associated with the request.
+     */
+    const SHIB_SESSION_ID = "Shib-Session-ID";
+
+    /**
+     * This is array of attributes which $this->authenticate()
+     * method should check for.
+     *
+     * WARNING: can contain only such attributes, which are writeable to user table!
+     *
+     * @var array attribsToCheck
+     */
+    protected $attribsToCheck = [
+        'cat_username', 'cat_password', 'email', 'lastname', 'firstname',
+        'college', 'major', 'home_library'
+    ];
+
     /**
      * Session manager
      *
      * @var \Laminas\Session\ManagerInterface
      */
     protected $sessionManager;
 
+    /**
+     * Configuration loading implementation
+     *
+     * @var ConfigurationLoadingInterface
+     */
+    protected $configurationLoading;
+
     /**
      * Constructor
      *
      * @param \Laminas\Session\ManagerInterface $sessionManager Session manager
      */
-    public function __construct(\Laminas\Session\ManagerInterface $sessionManager)
+    public function __construct(\Laminas\Session\ManagerInterface $sessionManager,
+        ConfigurationLoadingInterface $configurationLoading)
     {
         $this->sessionManager = $sessionManager;
+        $this->configurationLoading = $configurationLoading;
+    }
+
+    /**
+     * Get the URL to establish a session (needed when the internal VuFind login
+     * form is inadequate).  Returns false when no session initiator is needed.
+     *
+     * @param string $target Full URL where external authentication method should
+     * send user after login (some drivers may override this).
+     *
+     * @return bool|string
+     */
+    public function getSessionInitiator($target)
```

Comment:
I see that the order of some methods has been moved around in the file. It might be better to keep the existing method order, because it will make the diffs easier to review in the pull request.